src/Service/NotificationReceiverService.php

```diff
@@ -95,23 +94,23 @@ public function process(Request $requestObject)
         $basicAuthPassword = $requestObject->server->get('PHP_AUTH_PW');
 
         // Checks if notification is empty
-        if (empty($request)) {
+        if (empty($request) || empty($request['notificationItems'][0])) {
             $message = 'Notification is empty';
             $this->logger->critical($message);
-            return new JsonResponse(
-                [
-                    'success' => false,
-                    'message' => $message
-                ]
-            );
+            return new JsonResponse(['success' => false, 'message' => $message]);
         }
 
-        // Checks if notification is a test notification
-        $isTestNotification = $this->isTestNotification($request);
+        // First item in the notification
+        $firstNotificationItem = $request['notificationItems'][0]['NotificationRequestItem'];
 
         // Authorize notification
-        if (!$this->isAuthorized($isTestNotification, $basicAuthUser, $basicAuthPassword)) {
-            throw new AuthorizationException();
+        if (!$this->notificationReceiver->isAuthenticated(
+            $firstNotificationItem,
+            $this->configurationService->getMerchantAccount(),
```

Comment:
Should we start using the sales channel ID from the request object (`sw-sales-channel` I believe) to get the merchant account and not only the default one?

Reply:
@acampos1916 sure we can do that, we already have the variable in this method. Could that make a difference with this authentication?

Reply:
Ah yes, let's use that variable `$salesChannelId`. I also see that the `isAuthenticated()` function should receive the config user and pass and not the `$_SERVER` vars as the library method already does that. Maybe that's an old bug.